automata/fa/gnfa.py

```diff
@@ -320,39 +319,16 @@ def option(self):
     def reverse(self):
         raise NotImplementedError
 
-    def show_diagram(self, path=None, show_None=True):
-        """
-        Creates the graph associated with this DFA
-        """
-        # Nodes are set of states
-
-        graph = Dot(graph_type="digraph", rankdir="LR")
-        nodes = {}
-        for state in self.states:
-            if state == self.initial_state:
-                # color start state with green
-                initial_state_node = Node(state, style="filled", fillcolor="#66cc33")
-                nodes[state] = initial_state_node
-                graph.add_node(initial_state_node)
-            else:
-                if state == self.final_state:
-                    state_node = Node(state, peripheries=2)
-                else:
-                    state_node = Node(state)
-                nodes[state] = state_node
-                graph.add_node(state_node)
-        # adding edges
-        for from_state, lookup in self.transitions.items():
-            for to_state, to_label in lookup.items():  # pragma: no branch
-                if to_label is None and show_None:
-                    to_label = "ø"
-                    graph.add_edge(
-                        Edge(nodes[from_state], nodes[to_state], label=to_label)
-                    )
-                elif to_label is not None:
-                    graph.add_edge(
-                        Edge(nodes[from_state], nodes[to_state], label=to_label)
-                    )
-        if path:
-            graph.write_png(path)
-        return graph
+    def iter_transitions(self):
+        return (
+            (from_, to_, symbol)
+            for from_, lookup in self.transitions.items()
+            for to_, symbol in lookup.items()
+            if symbol is not None
+        )
+
+    def is_accepting(self, state):
```

Comment:
@eliotwrobson You know, I've been thinking about this ever since [I first asked](https://github.com/caleb531/automata/pull/129/files#r1153860767) about the need for `is_accepting` and realized that it was for things like GNFA.

Firstly, I would say that the [formal definition of a GNFA](https://en.wikipedia.org/wiki/Generalized_nondeterministic_finite_automaton#Formal_definition) stipulates that a GNFA has one and only one accept state, which I would prefer not to stray from. However, I think I have a solution that will work for everyone:

My proposal is to add a `@property`-type member to GNFA called `final_states` , which simply returns a `frozenset` containing that single final state. Something like:

```python
@property
def final_states(self):
    return frozenset({self.final_state})
```

The nice thing about this is because `frozenset` is hashable, `final_states` will always return something that is comparably the same on each access (hence why I didn't bother to suggest we use `@cached_property` or something).